crates/context/src/local.rs

```diff
@@ -0,0 +1,91 @@
+use bytecode::{CodeType, Eof};
+use context_interface::LocalContextTr;
+use primitives::{keccak256, Bytes, HashMap, B256};
+
+/// Local context that is filled by execution.
+#[derive(Clone, Debug, Default)]
+pub struct LocalContext {
+    /// Mapping of initcode hash that contains raw bytes ready for validation or status of validation.
+    ///
+    /// Used in EIP-7873 EOF - TXCREATE to fetch initcode by hash and cache its validation.
+    pub initcode_mapping: HashMap<B256, Initcode>,
+}
+
+impl LocalContextTr for LocalContext {
+    fn insert_initcodes(&mut self, initcodes: &[Bytes]) {
+        self.initcode_mapping = initcodes
+            .iter()
+            .map(|b| (keccak256(b), Initcode::new(b.clone())))
+            .collect();
+    }
+
+    fn clear(&mut self) {
+        self.initcode_mapping.clear();
+    }
+
+    fn get_validated_initcode(&mut self, hash: B256) -> Option<Bytes> {
+        let initcode = self.initcode_mapping.get_mut(&hash)?;
+        initcode.validate().cloned()
```

Comment:
This will mark intcode as `valid` or `invalid`, in case of Invalid None will be returned. Initial status is `PendingVerification`